docs/docs/general/03-Element Selection/recommended-practices.md

```diff
@@ -4,168 +4,242 @@ title: Recommended Practices
+
+<!-- vale off -->
+* Text linebreaks and text detection fails: See [Text Detection Fails Because of Linebreaks](#text-detection-fails-because-of-linebreaks)
+* Elements change shape or color during a workflow run: See [Color of Same Element Changes](#color-of-same-element-changes) and [Duplicate Elements](#duplicate-elements)
+<!-- vale oon -->
 
-### Avoid Optical Character Recognition (OCR) on Too Many Element
+---
 
-If you use `containsText()`, `withText()`, `withExactText()` or `withTextRegex()` **OCR** is applied to all elements detected on your screen. This can slow down AskUI. It is more efficient to narrow down the elements first. For example, if you want to click a `button` with a specific text you should select all buttons first.
+## Detection Is Flaky or Mislabeled
 
-```javascript
-// Do this
-await aui.click().button('See here').exec();
-await aui.clickText('Sign in');
+### Flaky Detection Across Runs
+Visual detection of elements is highly sensitive to changes in your UI, even if they are small. For example a button that you target may fail randomly on a workflow run every now and then:
 
-// And NOT this
-await aui.click('See here').exec();
-await aui.click('Sign in').exec();
+```typescript
+await aui.click().button().withText('Sign in').exec();
 ```
 
-### Avoid Custom Element Detection if Possible
+You can do three things about this:
 
-If you use `customElement()` you are doing an image-in-image search. Use this sparingly for elements that you can not detect as the execution time is sometimes slower!
+1. Make the element-description more specific, for example with [relational selectors](../../api/04-Relations/above.md)
+2. Add a custom element with `or()` as a fallback
+3. Combine the two methods
 
-```javascript
-// Do this
-await aui.clickButton('Login');
-await aui.clickText('Overview');
+All three of them you can find in the following code:
 
-// And NOT this
-await aui.click().customElement({
-  customImage: '.../login_button.png', 
-  name: 'login button',
-}).exec();
-await aui.click().customElement({
-  customImage: '.../text_overview.png', 
-  name: 'overview button',
-}).exec();
+```typescript
+// Make element-description more specific
+await aui.click().button().withText('Sign in')
+                 .leftOf().button().withText('Login')
+                 .exec();
+
+// Fallback to custom element
+await aui.click().button().withText('Sign in')
+                 .or()
+                 .customElement({
+                     customImage: 'custom_elements/signin.button.png'
+                   })
+                 .exec();
+
+// Add additional element-description and a custom element as fallback
+await aui.click().button().withText('Sign in');
+                 .or().button().containsText('Sign')
+                 .or().customElement({
+                     customImage: 'custom_elements/signin.button.png'
+                   })
+                 .exec();
 ```
```

Comment:
I would combine the code snippets directly with the corresponding workaround description so that the reader does not have to keep the whole block in mind or jump between blocks.